framework/src/mesh/MooseMesh.C

```diff
@@ -607,6 +610,27 @@ MooseMesh::nodeToElemMap()
   return _node_to_elem_map;
 }
 
+std::map<dof_id_type, std::vector<dof_id_type> > &
+MooseMesh::nodeToActiveElemMap()
+{
+  if (!_node_to_active_elem_map_built) // Guard the creation with a double checked lock
+  {
+    Threads::spin_mutex::scoped_lock lock(Threads::spin_mtx);
+    if (!_node_to_active_elem_map_built)
+    {
+      MeshBase::const_element_iterator       el  = getMesh().active_elements_begin();
```

Comment:
Then I don't think we should make an extra map for this.  We're essentially doubling the storage for most problems... and it's non "parallel" memory (meaning it won't go down as you spread the problem out).

If you need something specific for your problem, you can cache it locally in your object (just update it on meshChanged() ).

I was onboard when this was going to be something that was going to "scale"... but creating a second global map isn't the right idea.

Now: if you want to make this "activeSemiLocalNodeMap()" I would be on board and it would include ghosts ;-)